libs/ddhub-client-gateway-frontend/ui/channels/src/lib/containers/modals/Create/Details/Details.effects.ts

```diff
@@ -83,10 +86,26 @@ export const useDetailsEffects = (channelValues: ICreateChannel) => {
     ]);
     Object.entries(details).forEach(([name, value]) => {
       setValue(name, value);
+
+      if (name === 'connectionType') {
+        const isPublish = value === ConnectionType.Publish;
+        setShowEncryption(isPublish);
+      } else if (name === 'payloadEncryption') {
+        setIsEncrypt(!!value);
+      }
     });
     triggerValidation();
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [channelValues]);
 
-  return { fields, register, isValid, handleSubmit, control };
+  const connectionOnChange = (event: any) => {
```

Comment:
```suggestion
  const connectionOnChange = (event: Event) => {
```
Will work here?

Reply:
I tried `Event` and `ChangeEvent<HTMLInputElement>`, will throw error at `FormRadio` component.